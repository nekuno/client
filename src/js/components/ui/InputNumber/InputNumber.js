import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Input from '../Input/Input.js';
import styles from './InputNumber.scss';

export default class InputNumber extends Component {

    static propTypes = {
        title       : PropTypes.string,
        placeholder : PropTypes.string,
        defaultValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
        doNotFocus  : PropTypes.bool,
        doNotScroll : PropTypes.bool,
        onChange    : PropTypes.func,
        maxNum      : PropTypes.number,
        minNum      : PropTypes.number,
        size        : PropTypes.oneOf(['regular', 'small'])
    };

    constructor(props) {
        super(props);

        this.onChange = this.onChange.bind(this);

        this.state = {
            isValid: true
        }
    }

    onChange(value) {
        const {minNum, maxNum} = this.props;
        value = parseInt(value);

        if (value < minNum) {
            this.setState({isValid: false});
        } else if (value > maxNum) {
            this.setState({isValid: false});
        } else if (value) {
            this.setState({isValid: true});
            this.props.onChange(value)
        }
    }

    render() {
        const {isValid} = this.state;
        const textColor = isValid ? '' : 'red';
        return (
            <div className={styles.inputNumber}>
                <div className={styles.title + ' small'}>
                    {this.props.title}
                </div>
                <Input {...this.props} type={'number'} onChange={this.onChange} textColor={textColor}/>
            </div>
        );
    }
}