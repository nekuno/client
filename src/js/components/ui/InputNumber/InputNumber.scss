@import '../../../../scss/_partials/colors';

.input-number {
  .title {
    font-weight: 600;
  }
}