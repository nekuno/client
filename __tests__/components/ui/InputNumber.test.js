import React from 'react';
import InputNumber from '../../../src/js/components/ui/InputNumber/InputNumber.js';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test InputNumber component', () => {
    const mockFn = jest.fn();
    const inputNumber = shallow(<InputNumber placeholder={'Placeholder'} defaultValue={'Default value'} minNum={8} maxNum={80} onChange={mockFn}/>);

    it('should be defined', () => {
        expect(inputNumber).toBeDefined();
    });
    it('should render correctly', () => {
        expect(inputNumber).toMatchSnapshot();
    });
});
