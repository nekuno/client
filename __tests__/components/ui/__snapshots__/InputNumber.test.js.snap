// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputNumber component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputNumber
    defaultValue="Default value"
    maxNum={80}
    minNum={8}
    onChange={[MockFunction]}
    placeholder="Placeholder"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title small"
        />,
        <Input
          defaultValue="Default value"
          maxNum={80}
          minNum={8}
          onChange={[Function]}
          placeholder="Placeholder"
          textColor=""
          type="number"
        />,
      ],
      "className": "inputNumber",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "title small",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultValue": "Default value",
          "maxNum": 80,
          "minNum": 8,
          "onChange": [Function],
          "placeholder": "Placeholder",
          "textColor": "",
          "type": "number",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title small"
          />,
          <Input
            defaultValue="Default value"
            maxNum={80}
            minNum={8}
            onChange={[Function]}
            placeholder="Placeholder"
            textColor=""
            type="number"
          />,
        ],
        "className": "inputNumber",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "title small",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultValue": "Default value",
            "maxNum": 80,
            "minNum": 8,
            "onChange": [Function],
            "placeholder": "Placeholder",
            "textColor": "",
            "type": "number",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
